types/actioncable/index.d.ts

```diff
@@ -38,6 +38,7 @@ declare module ActionCable {
 
   function createConsumer(): Cable;
   function createConsumer(url: string): Cable;
+  function createConsumer(url: () => string): Cable;
```

Comment:
```suggestion
  function createConsumer(url?: string | (() => string)): Cable;
```

would be conventional (I thought there was a lint rule forbidding combinable overloads, in fact. Maybe it’s turned off for this package.)